Source/NSTableView.m

```diff
@@ -5019,9 +5035,18 @@ - (void) tile
 
 - (void) drawRow: (NSInteger)rowIndex clipRect: (NSRect)clipRect
 {
-  [[GSTheme theme] drawTableViewRow: rowIndex
-		   clipRect: clipRect
-		   inView: self];
+  if (_viewBased)
```

Comment:
Maybe we could put the horrible copied code to determine which columns need a redraw here, to avoid having all these copies?

Reply:
First get it working then optimize. 😃 that's always been my philosophy. 

I've been considering how to refactor this code since there is a ton of it in gsthemedrawing which doesn't seem to belong there.   We don't want to allow a theme developer to override the actual internal functionality of this.

Once I understand more about how this works on macOS I am going to pull the common parts out and refactor it so that there is no repeated/redundant code.

Reply:
Optimized. :)